sdk/core/azure-core/src/main/java/com/azure/core/implementation/util/ConfigurationUtils.java

```diff
@@ -0,0 +1,110 @@
+            throw new IllegalArgumentException("Duration can't be negative");
+        }
+
+        return Duration.ofMillis(timeoutMillis);
+    };
+
+    private static final Function<String, Integer> INTEGER_CONVERTER = (value) -> Integer.valueOf(value);
+    private static final Function<String, String> STRING_CONVERTER = (value) -> value;
+
+    /**
+     * Creates default {@link ConfigurationPropertyBuilder} configured to redact property value.
+     *
+     * @param name property name.
+     * @return instance of {@link ConfigurationPropertyBuilder}.
+     */
+    public static ConfigurationPropertyBuilder<String> stringSharedPropertyBuilder(String name) {
+        return new ConfigurationPropertyBuilder<>(name, STRING_CONVERTER).shared(true);
+    }
+
+    /**
+     * Creates {@link ConfigurationPropertyBuilder} configured to log property value and
+     * parse value using {@link Integer#valueOf(String)}, proxying {@link NumberFormatException} exception.
+     *
+     * @param name property name.
+     * @return instance of {@link ConfigurationPropertyBuilder}.
+     */
+    public static ConfigurationPropertyBuilder<Integer> integerSharedPropertyBuilder(String name) {
+        return new ConfigurationPropertyBuilder<>(name, INTEGER_CONVERTER).canLogValue(true).shared(true);
+    }
+
+    /**
+     * Creates {@link ConfigurationPropertyBuilder} configured to log property value and
+     * parses value as long number of milliseconds, proxying  {@link NumberFormatException} exception.
+     *
+     * @param name property name.
+     * @return instance of {@link ConfigurationPropertyBuilder}.
+     */
+    public static ConfigurationPropertyBuilder<Duration> durationSharedPropertyBuilder(String name) {
+        return new ConfigurationPropertyBuilder<>(name, DURATION_CONVERTER).canLogValue(true).shared(true);
+    }
+
+    /**
+     * Creates {@link ConfigurationPropertyBuilder} configured to log property value and
+     * parse value using {@link Boolean#parseBoolean(String)}.
+     *
+     * @param name property name.
+     * @return instance of {@link ConfigurationPropertyBuilder}.
+     */
+    public static ConfigurationPropertyBuilder<Boolean> booleanSharedPropertyBuilder(String name) {
+        return new ConfigurationPropertyBuilder<>(name, BOOLEAN_CONVERTER).canLogValue(true).shared(true);
+    }
+
+    /**
+     * Attempts to convert the configuration value to {@code T}.
+     *
+     * If the value is null or empty then the default value is returned.
+     *
+     * @param value Configuration value retrieved from the map.
+     * @param defaultValue Default value to return if the configuration value is null or empty.
+     * @param <T> Generic type that the value is converted to if not null or empty.
+     * @return The converted configuration, if null or empty the default value.
+     */
+    @SuppressWarnings("unchecked")
+    public static <T> T convertOrDefault(String value, T defaultValue) {
```

Comment:
This seems to handle only primitive types. Is there a plan to introduce object mapper here? If not we should probably change name and add docs explaining that `convertOrDefault("foo", new HashMap())` won't work.